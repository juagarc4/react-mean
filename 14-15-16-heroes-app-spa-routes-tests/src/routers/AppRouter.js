import React, { useContext } from 'react'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'

import { DashboardRoutes } from './DashboardRoutes'
import { LoginScreen } from '../components/login/LoginScreen'
import { PrivateRoute } from './PrivateRoute'
import { AuthContext } from 'auth/AuthContext'
export function AppRouter() {
  const { user } = useContext(AuthContext)
  return (
    <Router>
      <div>
        <Switch>
          <Route exact path="/login" component={LoginScreen} />
          <PrivateRoute
            path="/"
            component={DashboardRoutes}
            isAuthenticated={user.logged}
          />
        </Switch>
      </div>
    </Router>
  )
}
