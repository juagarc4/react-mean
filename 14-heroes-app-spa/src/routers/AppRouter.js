import React from 'react'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'

import { LoginScreen } from '../components/login/LoginScreen'
import { Navbar } from '../components/ui/Navbar'
import { MarvelScreen } from '../components/marvel/MarvelScreen'
export function AppRouter() {
  return (
    <Router>
      <div>
        <Navbar />
        <Switch>
          <Route exact path='/login' component={LoginScreen} />
          <Route exact path='/' component={MarvelScreen} />
        </Switch>
      </div>
    </Router>
  )
}
