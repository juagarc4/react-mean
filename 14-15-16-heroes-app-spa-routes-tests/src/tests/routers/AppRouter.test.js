import { AuthContext } from 'auth/AuthContext'
import { mount } from 'enzyme'
import { AppRouter } from 'routers/AppRouter'

describe('Tests ins <AppRouter/>', () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: { logged: false },
  }

  test('Should show login if not authenticate', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('Should show <MarvelScreen/> if authtenticated', () => {
    const contextValue = {
      dispatch: jest.fn(),
      user: { name: 'Raul', logged: true },
    }

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    )
    expect(wrapper.find('h1').text().trim()).toBe('Marvel Screen')
  })
})
