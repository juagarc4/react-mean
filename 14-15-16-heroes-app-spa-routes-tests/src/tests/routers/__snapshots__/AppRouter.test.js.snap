// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests ins <AppRouter/> Should show login if not authenticate 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login Screen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
